infra/core/env/scripts/Add-SshKeyToGitHubRepo.ps1

```diff
@@ -0,0 +1,213 @@
+            "Authorization"        = "Token $InstallationToken"
+            "Accept"               = "application/vnd.github+json"
+            "X-GitHub-Api-Version" = "2022-11-28"
+        }
+    
+        $keyTitle = "$($Environment.ToLower())_01"
+        $repoKeysUrl = "https://api.github.com/repos/defra/adp-flux-services/keys"
+        
+        Write-Debug "Reading all the deploy keys..."
+        [array]$keys = Invoke-RestMethod -Method Get -Uri $repoKeysUrl -Headers $headers
+        if ($keys) {
+            Write-Debug "Check if key already exists..."
+            [object]$existingKey = $keys | Where-Object { $_.title -eq $keyTitle }
+
+            if ($existingKey -and $existingKey -ne '') {
+                Write-Output "Deleting existing key '$keyTitle' having id '$($existingKey.id)'."
+                Invoke-RestMethod -Method Delete -Uri ("$repoKeysUrl/{0}" -f $existingKey.id) -Headers $headers | Out-Null
+            }
+        }
+
+        $body = @{
+            "title" = $keyTitle
+            "key" = $DeployKey
+            "read_only" = $false
+        } | ConvertTo-Json
+        Write-Output "Adding new key '$keyTitle'..."
+        Invoke-RestMethod -Method Post -Uri $repoKeysUrl -Body $body -Headers $headers | Out-Null
+    }
+
+    end {
+        Write-Debug "${functionName}:Exited"
+    }
+}
+
+Set-StrictMode -Version 3.0
+
+[string]$functionName = $MyInvocation.MyCommand
+[datetime]$startTime = [datetime]::UtcNow
+
+[int]$exitCode = -1
+[bool]$setHostExitCode = (Test-Path -Path ENV:TF_BUILD) -and ($ENV:TF_BUILD -eq "true")
+[bool]$enableDebug = (Test-Path -Path ENV:SYSTEM_DEBUG) -and ($ENV:SYSTEM_DEBUG -eq "true")
+
+Set-Variable -Name ErrorActionPreference -Value Continue -scope global
+Set-Variable -Name InformationPreference -Value Continue -Scope global
+
+if ($enableDebug) {
+    Set-Variable -Name VerbosePreference -Value Continue -Scope global
+    Set-Variable -Name DebugPreference -Value Continue -Scope global
+}
+
+Write-Host "${functionName} started at $($startTime.ToString('u'))"
+Write-Debug "${functionName}:AppIdSecretName=$AppIdSecretName"
+Write-Debug "${functionName}:AppKeySecretName=$AppKeySecretName"
+Write-Debug "${functionName}:Environment=$Environment"
+Write-Debug "${functionName}:SSHPublicKeySecretName=$SSHPublicKeySecretName"
+Write-Debug "${functionName}:KeyVaultName=$KeyVaultName"
+
+try {
+    Import-Module $PSHelperDirectory -Force
+
+    $appId = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $AppIdSecretName -AsPlainText
+    $appKey = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $AppKeySecretName -AsPlainText
+    $jwt = Get-GithubJwt -AppId $appId -AppKey $appKey
+
+    $installationToken = Get-InstallationToken -GitHubJwtToken $jwt
+
+    $deployKey = Get-AzKeyVaultSecret -VaultName $KeyVaultName -Name $SSHPublicKeySecretName -AsPlainText
+    Set-NewDeployKey -InstallationToken $installationToken -Environment $Environment -DeployKey $deployKey
+
+    $exitCode = 0
```

Comment:
Add more logging here?

Reply:
Changed to CLI call with PowerShell helpers that will add additional debug logging..